laser_slam/include/laser_slam/laser_track.hpp

```diff
@@ -47,6 +47,9 @@ class LaserTrack {
   /// \brief Get one local cloud in world frame.
   void getLocalCloudInWorldFrame(const Time& timestamp, DataPoints* out_point_cloud) const;
 
+  /// \brief Get all laser scans.
+  const std::vector<LaserScan>* getLaserScans() const;
```

Comment:
I'd would indeed recommend the const ref. Simply because if it is a pointer you would need to check as a caller whether it is not null!
Why does `const std::vector<LaserScan>& laserScans = ... ;` feel wrong? Btw. I'd also recommend `const auto && laserScans` instead.
There is another option, that might feel better: If you only need it a few times you can also just call `getLaserScans()` wherever you need it and don't store the reference. At least if the getter is inlineable there is no difference eventually.